src/models/Settings.js

```diff
@@ -303,7 +303,7 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, OAuth2Util, config) {
               click: function (e) {
                 e.preventDefault();
                 if (self.get('oauth2Enabled')) {
-                  OAuth2Util.getTokens(self, {idp: idp.id});
+                  OAuth2Util.getTokens(self, {idp: idp.id}); // TODO
```

Comment:
@haishengwu-okta @jmelberg-okta Question: this one does not have/pass a Controller. What you guys think we should use in this case?

Reply:
Given that OAuth errors can occur in a variety of cases, it might make sense to  map all OAuth errors to a controller named `oauth`. I'm thinking of the following scenarios:

1. User is going through the authentication/authorization flow for the first time:
    - The primary auth page is rendered and an error occurs because the user is **not** assigned to the application. While this happens on the primary auth page, we're technically redirecting back from Okta in an error state.
2. The user's token is no longer valid, and we attempt to renew it via `signIn.tokenManager.renew`. If an error occurs in that state, we technically **do not** have a controller present.

Reply:
iirc, `configuredButtons` are used at either primary auth or identity first auth page. and curious what is `this` context at this click function and if could get something out of `this`?

Reply:
Created https://github.com/okta/okta-signin-widget/pull/652 to move logic for creating button options.

@haishengwu-okta `this` context has a className parameter with this kind of value: `"social-auth-button social-auth-facebook-button`.
Unfortunately, `this` context does not have the controller.

Reply:
Updated https://github.com/okta/okta-signin-widget/pull/652 so we can have a controller :D